feature: native lambda deployment

oponomarov-tu opened the pull request:
This PRs: 
- adds `lambda_package_type` (defaults to `Image`) variable to control what runtime is used for executing Lambda: `Image` or `Zip`,
- moves some of the hardcoded values into variables.

bwhaley commented on modules/terraform-aws-alternat/variables.tf:
```diff
@@ -162,3 +162,45 @@ variable "vpc_id" {
+  description = "Environment variables to be provided to the lambda function."
+  type        = map(string)
+  default     = null
+}
+variable "lambda_zip_path" {
```
Nit: Needs line break.

bwhaley commented on modules/terraform-aws-alternat/variables.tf:
```diff
@@ -162,3 +162,45 @@ variable "vpc_id" {
   type        = string
 }
+
+variable "lambda_package_type" {
+  description = "The lambda deployment package type. Valid values are \"Zip\" and \"Image\". Defaults to \"Image\"."
+  type        = string
+  default     = "Image"
+  nullable    = false
+}
```
Nit: Add [validation](https://developer.hashicorp.com/terraform/language/values/variables#custom-validation-rules).

bwhaley commented on modules/terraform-aws-alternat/variables.tf:
```diff
@@ -121,7 +121,7 @@ variable "nat_instance_sg_name_prefix" {
 
 variable "nat_lambda_function_role_name" {
-  description = "Name ot use for the IAM role used by the replace-route Lambda function. Must be globally unique in this AWS account."
+  description = "Name to use for the IAM role used by the replace-route Lambda function. Must be globally unique in this AWS account."
```
👍

bwhaley commented on modules/terraform-aws-alternat/variables.tf:
```diff
@@ -162,3 +162,45 @@ variable "vpc_id" {
+  }
+}
+
+variable "lambda_environment_variables" {
```
Out of curiosity, what is the intended use for this?

oponomarov-tu replied:
I thought it would be nice to future-proof ourselves with an option to add custom environmental variables, for example to overwrite defaults (if more than one given map or object defines the same key or attribute, then the one that is later in the argument sequence takes precedence by `merge`) or supply some Python-specific configuration, like logging if there's going to be one, etc.

bwhaley reviewed (commented):
Nice improvement, thank you. I tested it out and worked well.

Just a couple of nits for you to consider.

oponomarov-tu commented:
> Nice improvement, thank you. I tested it out and worked well.
> 
> Just a couple of nits for you to consider.

Thanks, @bwhaley. I've addressed the comments in the last commit.
